drivers/pio/sunxi-pio.c

```diff
@@ -98,21 +93,14 @@ sunxi_pio_probe(struct device *dev)
 
 	/* Disable IO for all pins. */
 	for (size_t port = 0; port < MAX_PORTS; ++port) {
-		for (size_t pin = 0; pin < MAX_PINS_PER_PORT; pin += 8) {
-			mmio_write32(dev->regs +
-			             CONFIG_REG(SUNXI_PIO_PIN(port, pin)),
-			             0x77777777);
-		}
+		for (size_t pin = 0; pin < MAX_PINS_PER_PORT; pin += 8)
+			mmio_write32(dev->regs + CONFIG_REG(pin), 0x77777777);
 	}
 
 	/* Disable and clear all IRQs. */
 	for (size_t port = 0; port < MAX_PORTS; ++port) {
-		mmio_write32(dev->regs +
-		             INT_CONTROL_REG(SUNXI_PIO_PIN(port, 0)),
-		             0);
-		mmio_write32(dev->regs +
-		             INT_CONTROL_REG(SUNXI_PIO_PIN(port, 0)),
-		             0xffffffff);
+		mmio_write32(dev->regs + INT_CONTROL_REG(port), 0);
+		mmio_write32(dev->regs + INT_STATUS_REG(port), 0x1);
```

Comment:
These are wrong as written. This *can* work, since there's only one register of each type per port, but you'd need to update the macros at the top of the file to take a port number only, and not a pin number.

Reply:
Fixed.